src/components/LanguageChangeModal/index.jsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { Modal } from "antd";
+import i18n from "i18next";
+import { useTranslation } from "react-i18next";
+
+// in games, page must be reloaded to change language. This modal is created to warn user about it
+const LanguageChangeModal = ({
+  // boolean
+  modalVisibility,
+  setModalVisibility,
+  // string. lowercase language abbreviation
+  newLanguage,
+}) => {
+  const { t } = useTranslation();
+  return (
+    <Modal
+      title={t("languageChangeModal.title")}
+      visible={modalVisibility}
+      cancelText={t("languageChangeModal.cancelText")}
+      okText={t("languageChangeModal.okText")}
+      onOk={() => {
+        setModalVisibility(false);
+        // prevents language to change before modal closes
+        setTimeout(() => {
```

Comment:
Our cases are very different but when I see setTimeout, I was also using setTimeout for post submission message and according to the feedback from Louis, manually setting time for waiting users is not the ideal practice. You may apply [modal](https://ant.design/components/modal/) from ant design.